crates/sui-indexer-alt-jsonrpc/src/api/transactions/filter.rs

```diff
@@ -133,6 +140,74 @@ async fn all_transactions(
     Ok(refs)
 }
 
+/// Fetch a page of transaction digests from the given `checkpoint` (by sequence number). Fetches
+/// one more result than was requested to detect a next page.
+async fn by_checkpoint(
+    ctx: &Context,
+    page: &Page<u64>,
+    checkpoint: u64,
+) -> Result<Vec<(i64, Vec<u8>)>, RpcError<Error>> {
+    let Some(checkpoint) = ctx
+        .loader()
+        .load_one(CheckpointKey(checkpoint))
+        .await
+        .context("Failed to load checkpoint")?
+    else {
+        return Ok(vec![]);
+    };
+
+    let summary: CheckpointSummary = bcs::from_bytes(&checkpoint.checkpoint_summary)
+        .context("Failed to deserialize checkpoint summary")?;
+
+    let contents: CheckpointContents = bcs::from_bytes(&checkpoint.checkpoint_contents)
+        .context("Failed to deserialize checkpoint contents")?;
+
+    // Transaction sequence number bounds from the checkpoint
+    let cp_hi = summary.network_total_transactions;
+    let cp_lo = summary.network_total_transactions - contents.inner().len() as u64;
```

Comment:
I was initially a little surprised to remember than we don't have info like checkpoint tx lo and hi stored in some convenient tables but then thinking about the large number of checkpoints we have it probably makes sense to have these in kv format.